src/viadot/orchestration/prefect/tasks/supermetrics.py

```diff
@@ -0,0 +1,73 @@
+"""Task for downloading data from Hubspot API to a pandas DataFrame."""
+
+from typing import Any
+
+import pandas as pd
+from prefect import task
+
+from viadot.orchestration.prefect.exceptions import MissingSourceCredentialsError
+from viadot.orchestration.prefect.utils import get_credentials
+from viadot.sources import Supermetrics
+
+
+@task(retries=3, log_prints=True, retry_delay_seconds=10, timeout_seconds=60 * 60)
+def supermetrics_to_df(
+    query_params: dict,
+    config_key: str | None = None,
+    credentials_secret: str | None = None,
+) -> pd.DataFrame:
+    """
+    Retrieves data from Supermetrics and returns it as a pandas DataFrame.
+
+    This function queries the Supermetrics API using the provided query parameters and returns the data
+    as a pandas DataFrame. The function supports both configuration-based and secret-based credentials.
+
+    The function is decorated with a Prefect task, allowing it to handle retries, logging, and timeout behavior.
+
+    Parameters
+    ----------
+    query_params : dict
+        A dictionary containing the parameters for querying the Supermetrics API. These parameters
+        define what data to retrieve and how the query should be constructed.
+        
+    config_key : str, optional
+        The configuration key used to retrieve stored connection parameters and credentials for Supermetrics.
+        If provided, this key is used to fetch configuration settings from a configuration file or environment.
+        If not provided, the `credentials_secret` parameter must be used to supply credentials.
```

Comment:
This doesn't even make sense. Just copy the description from other sources/tasks.

Reply:
done